src/Concerns/Extendable.php

```diff
@@ -35,6 +40,71 @@ public static function hasExtend(string $name): bool
         return array_key_exists($name, static::$extends);
     }
 
+    /**
+     * Register a pipe to be applied before an expectation is checked.
+     */
+    public static function pipe(string $name, Closure $handler): void
+    {
+        self::$pipes[$name][] = $handler;
+    }
+
+    /**
+     * Register an interceptor that should replace an existing expectation.
+     */
+    public static function intercept(string $name, string|Closure $filter, Closure $handler): void
```

Comment:
If we typehint the `Closure` with a structure, we could probably remove the later `phpstan-ignore-next-line` calls.

Reply:
done!